compiler/rustc_interface/src/passes.rs

```diff
@@ -77,126 +74,6 @@ fn count_nodes(krate: &ast::Crate) -> usize {
     counter.count
 }
 
-pub use boxed_resolver::BoxedResolver;
-mod boxed_resolver {
-    use super::*;
-
-    pub struct BoxedResolver(Pin<Box<BoxedResolverInner>>);
```

Comment:
I'm happy every time the weird stuff added by Zoxc turns out to be overengineering and not legitimate complexity.
Carrying an extra lifetime around may be more verbose, but is unambiguously simpler.